auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackPasswordPrompt.java

```diff
@@ -114,6 +113,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     private void next(String email, final String password) {
         final FirebaseAuth firebaseAuth = mActivityHelper.getFirebaseAuth();
+        setIntent(getIntent().putExtras(mActivityHelper.getMergeFailedIntent()));
```

Comment:
I think this might be clearer if this were closer to where it is used on line 145